app/src/main/java/com/android/halanvendordiscovery/data/details/repo/DetailsRepositoryImpl.kt

```diff
@@ -0,0 +1,19 @@
+package com.android.halanvendordiscovery.data.details.repo
+
+import com.android.halanvendordiscovery.data.details.mapper.MerchantRemoteDomainMapper
+import com.android.halanvendordiscovery.data.details.service.DetailsService
+import com.android.halanvendordiscovery.domain.details.repo.DetailsRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DetailsRepositoryImpl @Inject constructor(private val service: DetailsService):
+    DetailsRepository {
+    override suspend fun getMerchants(vendorId: String) =
+        withContext(Dispatchers.IO) {
```

Comment:
from what i know `withContext()` is used to switch the context of the scope you're currently in , so this means that up until reaching the repository you are using different thread than **_Dispatchers.IO_**.

if that is true then why is that ?

if my assumption is wrong , this mean that this `withContext()` is redundant and you don't need it  because you've already stated (in another layer , usually viewModel/presentation) that you are using Dispatchers.IO.